src/main/java/io/confluent/idesidecar/restapi/proxy/clusters/strategy/DirectSchemaRegistryClusterStrategy.java

```diff
@@ -1,11 +1,39 @@
 package io.confluent.idesidecar.restapi.proxy.clusters.strategy;
 
+import io.confluent.idesidecar.restapi.connections.DirectConnectionState;
+import io.confluent.idesidecar.restapi.proxy.clusters.ClusterProxyContext;
+import io.vertx.core.MultiMap;
 import jakarta.enterprise.context.ApplicationScoped;
 
 /**
- * Strategy for processing requests and responses for a Confluent local Kafka cluster.
+ * Strategy for forwarding incoming REST requests to the Schema Registry specified on a direct
+ * connection.
  */
 @ApplicationScoped
 public class DirectSchemaRegistryClusterStrategy extends ClusterStrategy {
-  // No specific configuration for this strategy
+  private static final String TARGET_SR_CLUSTER_HEADER = "target-sr-cluster";
+
+  /**
+   * Constructs the headers for the proxied request, and add the authentication headers from the
+   * credentials, and the `target-sr-cluster` header set to the connection's SR cluster ID.
+   * @param context the context of the proxy request
+   * @return the headers to be used in the proxy request to the Schema Registry
+   */
+  @Override
+  public MultiMap constructProxyHeaders(ClusterProxyContext context) {
+    var headers = super.constructProxyHeaders(context);
+    if (context.getConnectionState() instanceof DirectConnectionState directConnectionState) {
+      var srConfig = directConnectionState.getSpec().schemaRegistryConfig();
+      if (srConfig != null) {
+        var credentials = srConfig.credentials();
+        if (credentials != null) {
+          credentials.httpClientHeaders().ifPresent(map -> map.forEach(headers::add));
+        }
+      }
+    }
```

Comment:
Should we throw an exception or log an error if `context.getConnectionState()` is not a `DirectConnectionState`?

Reply:
We're not doing that in [ConfluentCloudSchemaRegistryClusterStrategy](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/java/io/confluent/idesidecar/restapi/proxy/clusters/strategy/ConfluentCloudSchemaRegistryClusterStrategy.java), and I expect that's the case because [ClusterStrategyProcessor](https://github.com/confluentinc/ide-sidecar/pull/169/files#diff-9783c6e6d6f537d0c569f437e05cc5227a5e8944eccb79842ae963104f036cb9) is always using it correctly and therefore it should never occur at runtime.

That means that throwing an exception will only help during development, but maybe that's reason enough.

Reply:
Actually, if we want to add that, I think we should do that in a followup, since we'd also want to add it in multiple other places.

Reply:
Doing it in a follow-up sounds good to me. I wanted to make sure that we don't swallow any errors.